source/openqasm/openqasm3/printer.py

```diff
@@ -776,13 +815,18 @@ def visit_ClassicalAssignment(
         self.visit(node.rvalue, context)
         self._end_statement(context)
 
-    def visit_Pragma(self, node: ast.Pragma, context: PrinterState) -> None:
+    def visit_Annotation(self, node: ast.Annotation, context: PrinterState) -> None:
         self._start_line(context)
-        self.stream.write("#pragma {")
+        self.stream.write("@")
+        self.stream.write(node.keyword)
+        if node.command is not None:
+            self.stream.write(" ")
+            self.stream.write(node.command)
         self._end_line(context)
-        with context.increase_scope():
-            for statement in node.statements:
-                self.visit(statement, context)
+
+    @_maybe_annotated
+    def visit_Pragma(self, node: ast.Pragma, context: PrinterState) -> None:
```

Comment:
Interesting corner that I hadn't considered - annotated pragmas.

Reply:
I'm happy to update this PR to either allow or permit this.  At the moment, `pragma` is in `statement` in the ANTLR grammar, which automatically means it can be annotated, but I could also re-arrange that rule to forbid this, if your working group would prefer.

Reply:
I've updated this so that `ast.Pragma` can no longer take annotations and updated the parser to make `pragma` a global-only statement.  I didn't make any changes to the spec of the text - I don't know if you'd also like to do that in a separate PR.